h/schemas/base.py

```diff
@@ -73,6 +73,43 @@ def validate(self, data):
         return appstruct
 
 
+def remove_unknown_properties(data, schema):
+    """
+    Remove unknown properties from the data.
+
+    Remove all properites in the data that aren't in the schema.
+
+    :param data: The data to be validated
+    :param schema: The jsonschema schema to validate against
+    """
+
+    def remove_props(data, properties):
+        for prop in properties:
+            del data[prop]
+
+    data_properties = [(data, schema["properties"])]
+    # While there are still properties to be checked.
+    while data_properties:
+        data_d, schema_d = data_properties.pop()
+
+        extra_props = set(data_d.keys()) - set(schema_d.keys())
+        remove_props(data_d, extra_props)
+
+        extra_props = []
```

Comment:
For our current use case do we actually have any need to deal with nested fields? We're already leaving out handling for arrays amongst other things so we might as well leave that out if we don't need it.

Reply:
Lyza wanted to make it work with nested fields. I'm also in agreement with that but yes, we are leaving out arrays, and other property fields. I have noted this in the description now. Judging by your reaction to this dropping a property I will treat the property that incorrectly contains nested properties as an invalid type and not drop it from the data. I will however, leave the if "properties" key check in place so that if someone runs this without validating it, it won't fail with an unhelpful key error.